### DISCUSSION TITLE
No "Start Successful" Discord notification on auto-update

### DISCUSSION THREAD

**Discussion by pwlgrzs** [mbround18/valheim-docker#261] [Help!]
Server stops at 5AM for auto-update following these variables:
  -e AUTO_UPDATE=1
  -e AUTO_UPDATE_SCHEDULE="0 5 * * *" 

Our Discord gets 2 notifications:

![image](https://user-images.githubusercontent.com/7893951/110086228-66e65380-7d92-11eb-879c-4ee6f192250a.png)

but we do not receive Start Successful message.

**Comment by CosmicHorrorDev** [comment]
Can you post a (censored) docker compose file (or equivalent), along with a server log and when you last pulled the image. This has been a bug that seems to have hit a few people, but I still haven't been able to reproduce.

Don't list the full webhook url you pass in, but knowing if it's discord or discordapp would be nice too.

**Comment by pwlgrzs** [comment]
`#!/bin/bash

docker stop valheim
docker rm valheim
docker pull mbround18/valheim:latest

docker run -d \
  --name valheim \
  -p 2456:2456/udp \
  -p 2457:2457/udp \
  -p 2458:2458/udp \
  -v /local/valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim \
  -v /local/valheim/server:/home/steam/valheim \
  -v /local/valheim/backup:/home/steam/backups \
  -e PUBLIC=1 \
  -e PORT=2456 \
  -e NAME= \
  -e PASSWORD="" \
  -e TZ=Europe/Warsaw \
  -e WORLD=World \
  -e AUTO_UPDATE=1 \
  -e AUTO_UPDATE_SCHEDULE="0 5 * * *" \
  -e UPDATE_ON_STARTUP=1 \
  -e AUTO_BACKUP=1 \
  -e AUTO_BACKUP_SCHEDULE="*/15 * * * *" \
  -e AUTO_BACKUP_REMOVE_OLD=1 \
  -e AUTO_BACKUP_DAYS_TO_LIVE=3 \
  -e AUTO_BACKUP_ON_UPDATE=1 \
  -e AUTO_BACKUP_ON_SHUTDOWN=1 \
  -e WEBHOOK_URL=https://discord.com/api/webhooks/ \
  -e WEBHOOK_BROADCAST_MESSAGE="HEHE VALHEIM" \
  --restart=always \
  mbround18/valheim`
  
  Image pulled: 2021-03-02 10:03:56
 Log:
[valheim_server.log](https://github.com/mbround18/valheim-docker/files/6089095/valheim_server.log)

**Comment by pwlgrzs** [comment]
Also, message for auto-update says: "Notification From: Your Valheim Server" while if I do odin stop it will say "Notification From: ACTUAL SERVER NAME".

**Comment by CosmicHorrorDev** [comment]
Does stopping and starting the container send out the messages correctly. I would pull the latest image, I think there was some work towards fixing this issue earlier today.

> Also, message for auto-update says: "Notification From: Your Valheim Server" while if I do odin stop it will say "Notification From: ACTUAL SERVER NAME".

This is a known issue. I'll go ahead and push out a fix for this.

**Comment by pwlgrzs** [comment]
I just pulled newest image but obviously we gotta wait till 5AM to see.

**Comment by CosmicHorrorDev** [comment]
No worries, gives me time to stall :D

In the meantime I went ahead and opened a pull request to fix the name not getting read for auto updates (#256)

![image](https://user-images.githubusercontent.com/30302768/110090303-215f6180-7d65-11eb-88f5-a38cb0aaf679.png)

**Reply by TheDakk** [reply]
Will the fix the issue with the auto update saying Your Server as well?

**Reply by CosmicHorrorDev** [reply]
That's the one. Names should show up correctly in cron jobs now 👍

**Reply by pwlgrzs** [reply]
All good!

**Reply by TheDakk** [reply]
This is still not resolved. The webhook is never sent after the auto update and it still says "Your Server"

![notifications](https://user-images.githubusercontent.com/79046704/110242932-3505f600-7f26-11eb-9bd3-a73b859d54a0.PNG)

[auto-update.log](https://github.com/mbround18/valheim-docker/files/6097272/auto-update.log)

**Reply by pwlgrzs** [reply]
Are you sure you got newest image? It's OK for me for another day.

**Reply by TheDakk** [reply]
I just updated yesterday

**Reply by TheDakk** [reply]
I should add it works on a cold startup and shutdown, just not during auto update.

**Reply by CosmicHorrorDev** [reply]
This should definitely be resolved. Are you sure you actually pulled the `lastest` image?

Edit: And the behavior you described about regular startup and shutdown still tracks for the old issue which was resolved

**Reply by TheDakk** [reply]
I think so? I run `docker-compose pull` and it says it’s downloading Valheim. Is there a way I can tell? I posted the auto update log and I never see the webhook sending on startup.

**Reply by pwlgrzs** [reply]
@TheDakk I think running "docker images mbround18/valheim" command would show what images you got

**Reply by TheDakk** [reply]
Thanks, here is what I'm running:

REPOSITORY          TAG       IMAGE ID       CREATED        SIZE
mbround18/valheim   latest    03d5efed78fd   18 hours ago   764MB
mbround18/valheim   <none>    925236c67758   46 hours ago   764MB
mbround18/valheim   <none>    a503b9fd465b   2 days ago     764MB
mbround18/valheim   <none>    c671a1a1b218   5 days ago     746MB
mbround18/valheim   <none>    3bea942a6411   8 days ago     746MB
mbround18/valheim   <none>    05fc72b6c8cc   2 weeks ago    495MB
mbround18/valheim   <none>    f44f94b6104a   2 weeks ago    660MB

**Reply by pwlgrzs** [reply]
I also have 03d5efed78fd, so indeed you got that image loaded. Could you share your compose file (redact any credentials etc.)

**Reply by TheDakk** [reply]
version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    restart: always
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME=
      - WORLD=
      - PASSWORD=
      - TZ=America/New_York
      - PUBLIC=1
      - AUTO_UPDATE=1
      - AUTO_UPDATE_SCHEDULE="0 6 * * *"
      - PUID=1000
      - PGID=998
      - AUTO_BACKUP=1
      - AUTO_BACKUP_SCHEDULE="*/30 * * * *"
      - AUTO_BACKUP_REMOVE_OLD=1
      - AUTO_BACKUP_DAYS_TO_LIVE=1
      - AUTO_BACKUP_ON_UPDATE=1
      - AUTO_BACKUP_ON_SHUTDOWN=1
      - WEBHOOK_URL=
    volumes:
    - ./valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim
    - ./valheim/server:/home/steam/valheim
    - ./valheim/backups:/home/steam/backups

**Reply by CosmicHorrorDev** [reply]
There should also be a message about running an outdated image when the container is started, but you can manually check the sha for the github commit the container is from since it's stored in `/home/steam/.version` with something like

```bash
docker exec $CONTAINER_NAME cat /home/steam/.version
```

Mine currently lists `3e2c851a7f8430194b4ca5960403e8c6ca2b28e6`, but there was a new release relatively recently.

**Reply by TheDakk** [reply]
CONTAINER ID   IMAGE          COMMAND                  CREATED      STATUS 
dbc1d3ed142b   c671a1a1b218   "/bin/bash /entrypoi…"   5 days ago   Up 11 hours (healthy) 

the sha is "f19ecd3d14030bafea42781d8744b774a0864f5c"

Looks like mine is running an older one somehow??


Do you need to run `docker-compose up` after you do a pull? I usually did a reboot which auto started the container. After running the compose up it shows the correct image now:


dockeruser@valheim-server-nyc3-01:~$ docker ps
CONTAINER ID   IMAGE                      COMMAND                  CREATED         STATUS
130667d7cb35   mbround18/valheim:latest   "/bin/bash /entrypoi…"   6 seconds ago   Up 4 seconds (health: starting)

**Reply by CosmicHorrorDev** [reply]
Honestly i'm not the most familiar with docker compose, but I believe that

```bash
docker-compose down
docker-compose pull
docker-compose up
```

Should update the container appropriately. There was a recent change allowing for checking for updates without restarting, so don't be surprised if you don't get a daily shutdown, startup message.
